packages/react-composites/src/composites/CallWithChatComposite/adapter/TestUtils.ts

```diff
@@ -0,0 +1,294 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+import {
+  Call,
+  CallAgent,
+  CallAgentOptions,
+  CallAgentKind,
+  CallFeature,
+  CallFeatureFactory,
+  CollectionUpdatedEvent,
+  DeviceManager,
+  GroupLocator,
+  IncomingCallEvent,
+  JoinCallOptions,
+  StartCallOptions,
+  TeamsMeetingLinkLocator,
+  VideoDeviceInfo
+} from '@azure/communication-calling';
+/* @conditional-compile-remove(unsupported-browser) */
+import { EnvironmentInfo } from '@azure/communication-calling';
+import {
+  CommunicationTokenCredential,
+  CommunicationUserIdentifier,
+  CommunicationUserKind,
+  PhoneNumberIdentifier,
+  PhoneNumberKind,
+  MicrosoftTeamsUserKind,
+  UnknownIdentifier,
+  UnknownIdentifierKind
+} from '@azure/communication-common';
+/* @conditional-compile-remove(calling-beta-sdk) */
+import { GroupChatCallLocator, MeetingLocator, RoomLocator, PushNotificationData } from '@azure/communication-calling';
+import {
+  CallState,
+  CallClientState,
+  StatefulCallClient,
+  createStatefulCallClient,
+  CallErrors,
+  CreateViewResult
+} from '@internal/calling-stateful-client';
+import EventEmitter from 'events';
+/**
+ *
+ */
+export class MockCallClient {
```

Comment:
We don't have a mock client implementation before? If so, this is a nice one! Have you tried this one:
https://jestjs.io/docs/es6-class-mocks#replacing-the-mock-using-mockimplementation-or-mockimplementationonce

so this will allow you only mock those functions really need to be implemented for your code

Reply:
It looks like he have written similar implementations, not exactly the same, but they are in the statefulclient package.